src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,777 @@ subroutine eigencalving(domain, err)
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+      type (block_type), pointer :: block
+      type (mpas_pool_type), pointer :: geometryPool
+      type (mpas_pool_type), pointer :: meshPool
+      type (mpas_pool_type), pointer :: velocityPool
+      type (mpas_pool_type), pointer :: scratchPool
+      real(kind=RKIND), pointer :: config_damage_stiffness_min
+      logical, pointer :: config_damage_rheology_coupling
+      logical, pointer :: config_print_calving_info
+      character, pointer :: config_damage_gl_setting
+      real (kind=RKIND), dimension(:), pointer :: damage
+      real (kind=RKIND), dimension(:), pointer :: damageMax
+      real (kind=RKIND), dimension(:), pointer :: damage_nye
+      real (kind=RKIND), dimension(:), pointer :: stiffnessFactor
+
+
+      integer, dimension(:), pointer :: cellMask
+      integer, dimension(:), pointer :: nEdgesOnCell ! number of cells that border each cell
+      integer, dimension(:,:), pointer :: cellsOnCell ! list of cells that neighbor each cell
+      integer, pointer :: nCells 
+      integer :: iCell, jCell, iNeighbor, n_damage_downstream
+      real(kind=RKIND) :: damage_downstream
+
+      err = 0
+
+
+      call mpas_pool_get_config(liConfigs, 'config_damage_stiffness_min', config_damage_stiffness_min)
+      call mpas_pool_get_config(liConfigs, 'config_damage_rheology_coupling', config_damage_rheology_coupling)
+      call mpas_pool_get_config(liConfigs, 'config_damage_gl_setting', config_damage_gl_setting)
+
+      ! block loop
+      block => domain % blocklist
+      do while (associated(block))
+
+         ! get pools
+         call mpas_pool_get_subpool(block % structs, 'geometry', geometryPool)
+         call mpas_pool_get_subpool(block % structs, 'velocity', velocityPool)
+         call mpas_pool_get_subpool(block % structs, 'mesh', meshPool)
+
+         ! get fields
+         call mpas_pool_get_dimension(meshPool, 'nCells', nCells)
+
+         call mpas_pool_get_array(meshPool, 'nEdgesOnCell', nEdgesOnCell)
+         call mpas_pool_get_array(meshPool, 'cellsOnCell', cellsOnCell)
+         call mpas_pool_get_array(meshPool, 'dvEdge', dvEdge)
+         call mpas_pool_get_array(geometryPool, 'cellMask', cellMask)
+         call mpas_pool_get_array(geometryPool, 'damage', damage)
+         call mpas_pool_get_array(geometryPool, 'damageMax', damageMax)
+         call mpas_pool_get_array(geometryPool, 'damage_nye', damage_nye)
+         call mpas_pool_get_array(velocityPool, 'stiffnessFactor', stiffnessFactor)
+
+
+         do iCell = 1, nCells
+            if (damage(iCell) < damageMax(iCell)) then
+                damage(iCell) = damageMax(iCell)
+            end if
+         end do
+         ! put the damageMax value back to restore the damage value (no heal)
+
+         where (damage < 0.0_RKIND)
```

Comment:
Now I also add the rheology coupling piece of code in the calculate_damage subroutine. You were right. We don't want to have the rheology coupling only when we maintain the broken ice (changed to preserve_high_damage already).

The preserve_high_damage function after advection is optional, i.e., we can choose if we want to let the damaged ice heal or not. So I don't agree it is the second phase of damage calculation.

Reply:
It seemed to me that the part of that routine that deals with the damage value near the GL should happen regardless of which healing option is chosen.  This is because during advection the first row of floating cells will have had garbage values advected in from the grounded ice.

Reply:
Thinking about it more, I also think the rheology coupling needs to occur after advection.  If you put it before, then the adjustments to the stiffnessParameter will not be consistent with the final state of the damage field.

One other option is to do all the damage evolution after advection occurs instead of some before and some after.  (In other words, make one damage routine that only gets called after advection.)  @stephenprice , do you have any thoughts about what order of operations is most appropriate?

Reply:
I think it is a good idea to put damage_calculation after advection. At the begining we can set damage = 0 in landice_grid.nc, and after advection, damage is also 0 before we update it in the damage calculation subroutine. In this case, we can put rheology coupling in the damage calculation, followed by the preserve_high_damage_subroutine